contracts/p1/StRSR.sol

```diff
@@ -158,6 +158,8 @@ abstract contract StRSRP1 is Initializable, ComponentP1, IStRSR, EIP712Upgradeab
     uint48 private lastWithdrawRefresh; // {s} timestamp of last refresh() during withdraw()
     uint192 public withdrawalLeak; // {1} gov param -- % RSR that can be withdrawn without refresh
 
+    // NOTE for future: Do not add a uint48 (or smaller) here
```

Comment:
Why? Seems fine to add. `withdrawalLeak` only takes up the first 20 bytes of the 32-byte slot, right?

Reply:
Is there a danger round upgrades when using a slot partially, that I don't know about?

Reply:
Storage layout is right-to-left, so it's preferred to isolate them between upgrades. (Although, it doesn't technically break it, just packs variables from two different versions)

Reply:
I'm failing to see why right-to-left or left-to-right would have any bearing, as long as its consistent. 

On the other hand obviously I can see intuitively why there might be a preference for isolating slots between upgrades...but it seems like when you actually go down to the low-level it shouldn't matter

Reply:
Yeah, it most certainly a preference thing and not really a requirement per-se. I can drop this if you think this is unnecessary.

Reply:
I think I'd like to keep the `using Bytes32Set` change (clarifying) but remove this one comment (confusing)

Reply:
then LGTM for merge

Reply:
actually the tests will fail without an update to the testing files for 3.0.0, but that is going in as part of #833